src/lib/containers/entity_finder/EntityFinder.tsx

```diff
@@ -40,6 +41,20 @@ const DEFAULT_VISIBLE_TYPES = [EntityType.PROJECT, EntityType.FOLDER]
 // This is necessary because undefined is returned by map.get when the item is not in the map
 export const NO_VERSION_NUMBER = -1
 
+const searchForOnlyTypesBooleanQuery = (
+  entityTypes: EntityType[],
+): KeyValue[] => {
+  // Boolean query terms will be combined with AND, and there's no way to OR.
+  // So we will negate searching for all omitted types.
+  const allTypes = Object.values(EntityType)
+  const typesToOmit = allTypes.filter(type => !entityTypes.includes(type))
+  return typesToOmit.map(type => ({
+    key: 'node_type',
+    value: type.toString(),
+    not: true,
+  }))
+}
+
```

Comment:
Our search API doesn't have a logical "OR" option, but we can negate searching for all other types, as long as our EntityType enum is kept up-to-date.